rust/lance/src/dataset/scanner.rs

```diff
@@ -124,27 +124,27 @@ impl ColumnOrdering {
 ///   .sum()
 /// ```
 pub struct Scanner {
-    dataset: Arc<Dataset>,
+    pub(crate) dataset: Arc<Dataset>,
```

Comment:
I'm not sure we want to make these `pub(crate)`. Perhaps we should move the `TableProvider` implementation into this file, or a submodule of it?